### ISSUE TITLE
Access to storage.cloud.google.com

### ISSUE THREAD

**Issue by Pupix** [googleapis/nodejs-storage#154]
Hello!

I hope this is the right repository to ask this question so here we go.

I am trying to implement custom authentication for my resources on Google Cloud Storage using this module with a service account. I am trying to abstract away the need for a Google account for my end users.

What the idea workflow would look like:
1. User queries `https://cdn.example.com/[[BUCKET]]/[[FILENAME]]` using Bearer token 
2. The API on that end intercepts the bucket and file name and checks the validity of the token
3. The API then would request the resource at `https://storage.cloud.google.com/[[BUCKET]]/[[FILENAME]]`
4. `Step 3` Returns the `Location` header that I will pass over to the user

After snooping around a bit I found out that the `Location` header returned in `step 3` in the form of `https://[[DATA]].googleusercontent.com/download/storage/v1/b/[[BUCKET]]/o/[[FILENAME]]?qk=[[KEY]]` is a public link that can be accessed by anonymous users too. Which is exactly what I want. However while using the storage API I can only see `selfLink` and `mediaLink`, not the link above.

I tried using [`google-auto-auth`](https://www.npmjs.com/package/google-auto-auth) to sign the request with my service account towards the `storage.cloud.google.com` endpoint but I get an `Unauthorized` error.

From looking [here](https://cloud.google.com/storage/docs/access-control/cookie-based-authentication) I understand that to access the `storage.cloud.google.com` is based on cookie authentication, which `google-auto-auth` doesn't seem to do. All it does is add a Bearer token to the header of the request.

**This looks like you need [signedUrls](https://cloud.google.com/storage/docs/access-control/signed-urls)**
Yea, that would be great, if it didn't expose the email of the service account.

Cheers!

TL;DR How to get the redirect URL from `storage.cloud.google.com` links using a service account?

**Comment by stephenplusplus** [comment]
I think these questions are more appropriate for someone on the Storage team, or the wider community of users themselves. You can get links to both from the Support page: https://cloud.google.com/storage/docs/support.

**Comment by Pupix** [comment]
Alright, posted it [here](https://stackoverflow.com/questions/49299116/access-to-storage-cloud-google-com-with-a-service-account) in case someone wants to help out.
